### DISCUSSION TITLE
data routes empty

### DISCUSSION THREAD

**Discussion by dgoh-zippia** [vercel/next.js#17643] [Help]
Running into this error when running next build as I am trying to implement a dynamic SSG page. After generating the _next folder, there's a routes-manifest.json file. Inside the json file, there should be a dataRoutes key with array value, but for some reason, mine appears empty which caused this error. My pages directory looks like this. 

```
pages
  |-- .[criteria]
         | -- blog.jsx
```

Stack trace:

> TypeError: Cannot read property 'replace' of undefined
    at routesManifest.dataRoutes.map.page (/mnt/c/Users/dgoh/Documents/GitHub/something/project/node_modules/next/dist/build/index.js:19:573)
    at Array.map (<anonymous>)
    at build (/mnt/c/Users/dgoh/Documents/GitHub/something/project/node_modules/next/dist/build/index.js:19:89)
    at process._tickCallback (internal/process/next_tick.js:68:7)

I am using a custom server if that has anything to do with it.

**Comment by Heedster** [comment]
I also have the same issue in the exact same line and word, but I am using vercel. 


```
TypeError: Cannot read property 'replace' of undefined
--
12:22:21.523 | at /vercel/workpath0/node_modules/next/dist/build/index.js:19:573
12:22:21.523 | at Array.map (<anonymous>)
12:22:21.523 | at build (/vercel/workpath0/node_modules/next/dist/build/index.js:19:89)
12:22:21.523 | at runMicrotasks (<anonymous>)
12:22:21.523 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

@dgoh-zippia Did you happen to figure out what it is?

The wierdest thing is that my build succeeds in local, and breaks in vercel. 

I think it should have something to do with filenames in linux and mac systems, but I have no clue what.

**Reply by Heedster** [reply]
I have figured out the issue I had, and it is extremely weird.

So, this happened after a particular commit, so, I removed things until the bug disappeared and figured out the cause.
I had added a new library ("recordrtc" in my case) in a new component in an existing page, and when I remove the usage of it, it seems to work. When I dynamically load that library using an import(), it seems to be working fine as well. So that solves it.

I have zero idea why this has resulted in this issue. I am regarding it as one of life's unanswerable questions. 
The line that breaks the compilation is related to dynamicRoutes and it resolving namedRoute regex. I have not added a route, not removed any for fixing this. I have zero idea why adding a new library in an existing SSR page has caused an issue within the routing logic.

If it works, it works! Hope this helps someone.

**Comment by DavidForss** [comment]
Same problem here!
It occurs for me when i add `target: 'serverless'` to my next.config.js-file.
I need that line to be able to use next-on-netlify and deploy on netlify.
Any thoughts appreciated!
